dockerfile/rocm5.0-pytorch1.9.0.dockerfile

```diff
@@ -0,0 +1,134 @@
+    echo -e "root soft nofile 1048576\nroot hard nofile 1048576" >> /etc/security/limits.conf
+
+# Install OFED
+ENV OFED_VERSION=5.2-2.2.3.0
+RUN cd /tmp && \
+    wget -q http://content.mellanox.com/ofed/MLNX_OFED-${OFED_VERSION}/MLNX_OFED_LINUX-${OFED_VERSION}-ubuntu18.04-x86_64.tgz && \
+    tar xzf MLNX_OFED_LINUX-${OFED_VERSION}-ubuntu18.04-x86_64.tgz && \
+    PATH=/usr/bin:${PATH} MLNX_OFED_LINUX-${OFED_VERSION}-ubuntu18.04-x86_64/mlnxofedinstall --user-space-only --without-fw-update --force --all && \
+    rm -rf MLNX_OFED_LINUX-${OFED_VERSION}*
+
+# Install ucx
+RUN cd /tmp && wget https://github.com/openucx/ucx/releases/download/v1.12.0/ucx-1.12.0.tar.gz && \
+    tar xzf ucx-1.12.0.tar.gz && \
+    cd ucx-1.12.0 && \
+    mkdir build && cd build/ && \
+    ../contrib/configure-release --prefix=/opt/ucx && \
+    make -j $(nproc) && make install && \
+    rm -rf /tmp/ucx-1.12.0
+
+# Install OpenMPI
+ENV OPENMPI_VERSION=4.0.5
+RUN cd /tmp && \
+    wget -q https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-${OPENMPI_VERSION}.tar.gz && \
+    tar xzf openmpi-${OPENMPI_VERSION}.tar.gz && \
+    cd openmpi-${OPENMPI_VERSION} && \
+    ./configure --enable-orterun-prefix-by-default && \
+    make -j $(nproc) all && \
+    make install && \
+    ldconfig && \
+    rm -rf /tmp/openmpi-${OPENMPI_VERSION}*
+
+# Install HPC-X
+RUN cd /opt && \
+    wget -q https://azhpcstor.blob.core.windows.net/azhpc-images-store/hpcx-v2.8.3-gcc-MLNX_OFED_LINUX-${OFED_VERSION}-ubuntu18.04-x86_64.tbz && \
+    tar xf hpcx-v2.8.3-gcc-MLNX_OFED_LINUX-${OFED_VERSION}-ubuntu18.04-x86_64.tbz && \
+    ln -s hpcx-v2.8.3-gcc-MLNX_OFED_LINUX-${OFED_VERSION}-ubuntu18.04-x86_64 hpcx && \
+    rm hpcx-v2.8.3-gcc-MLNX_OFED_LINUX-${OFED_VERSION}-ubuntu18.04-x86_64.tbz
+
+# Install Intel MLC
+RUN cd /tmp && \
+    mkdir -p mlc && \
+    cd mlc && \
+    wget --user-agent="Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0" https://www.intel.com/content/dam/develop/external/us/en/documents/mlc_v3.9a.tgz && \
+    tar xvf mlc_v3.9a.tgz && \
+    cp ./Linux/mlc /usr/local/bin/ && \
+    cd /tmp && \
+    rm -rf mlc
+
+# Install rccl-rdma-sharp-plugins
+ENV SHARP_VERSION=5.0
+RUN cd /opt/rocm && \
+    git clone -b release/rocm-rel-${SHARP_VERSION} https://github.com/ROCmSoftwarePlatform/rccl-rdma-sharp-plugins.git && \
+    cd rccl-rdma-sharp-plugins && \
+    ./autogen.sh && ./configure && make -j64
+
+# Install llvm+clang, required by hipify
+ENV LLVM_VERSION=13.0.0
+RUN cd /tmp && git clone -b llvmorg-${LLVM_VERSION} https://github.com/llvm/llvm-project.git && \
+    cd llvm-project/&& \
+    mkdir build && cd build && \
+    cmake -DCMAKE_INSTALL_PREFIX=/usr/lib/llvm-${LLVM_VERSION} -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLLVM_ENABLE_PROJECTS="clang"  -DCMAKE_BUILD_TYPE=Release ../llvm && \
+    make -j $(nproc) install && \
+    cd /tmp && rm -rf llvm-project
+
+# Install hipify
+ENV ROCM_VER=rocm-5.0.0
```

Comment:
Why we need sharp plugin, LLVM and Hipfy only in ROCm 5.0?

Reply:
in the previous version, hipify-perf was built-in, but it disappeared in rocm5.0

Reply:
Therefore, we need to ask AMD on this.

Reply:
they suggested to use 'apt install clang-hipify'